src/light_data/mod.rs

```diff
@@ -104,6 +104,18 @@ impl LightData {
         x.to_le_bytes()[..Self::byte_count(x) as usize].to_vec()
     }
 
+    /// Returns the `usize` that's represented by a little endian byte array.
+    ///
+    /// # Panics
+    ///
+    /// This function panics if the input array is longer than 4 bytes on 32-bit systems or
```

Comment:
Do we really want to panic when deserializing bad data (or even data that maybe 'good' on a different system? I think this function should probably return an `Option` or `Result` so failure can be propagated as a deserialization error rather than panicking.

Reply:
Yeah, Github seems to have eaten a comment I made about this in the approval above.